### PR TITLE
SCUMM: HE: support BiDi in hebrew fan translations

### PR DESCRIPTION
**Pull request by BLooperZ**
This adds support for writing BiDi text in upcoming hebrew fan translations of HE games.
It important to note that this check will also catch some officially translated games.
the official translation were only in audio and graphics, with the english text left intact (subtitles are hidden by default).

(the upcoming fan translation are text based)
convertBiDiString is used so the hebrew versions with english texts still get displayed in english.

still need to consider if the change in text alignment affects the existing translations, though.

Thanks

### CODE REVIEW

**Comment by BLooperZ** [comment]
Seems the localization credits are affected, need to find a way to differentiate them to disable BiDi functionality when playing these games.

**Comment by BLooperZ** [comment]
Fixed original translations by opting out using feature flag `GF_HE_NO_BIDI`

File: engines/scumm/string.cpp [hunk 1]
```diff
@@ -572,6 +573,31 @@ bool ScummEngine::newLine() {
...
+void ScummEngine_v72he::fakeBidiString(byte *ltext, bool ignoreVerb) const {
+	if (*ltext == 0x7F) {
+		ltext++;
+		while (*(ltext++) != 0x7F);
+	}
+	byte *loc = ltext;
+	byte tmp = 0;
+	while (1) {
+		while (*loc && *loc != 13) {
+			loc++;
```
Comment (bluegr) [inline engines/scumm/string.cpp:586]: You could use forEachLine() inside Common::String to avoid all of this string manipulation per line

Comment (BLooperZ) [reply engines/scumm/string.cpp:586]: Thanks,
tried look into it, but it seens the current implementation of `forEachLine` is not suitable for this as it
hard-coded to use `\n` (10) as new-line token
and require to use a `Common::String` object which might not be stable long-term for modifying the underlying buffer
(this implementation reverses the strings in-place)

**Comment by sev-** [comment]
Thanks!

**Pull request merged**